src/Symfony/Component/Serializer/Normalizer/AbstractNormalizer.php

```diff
@@ -302,21 +302,8 @@ protected function getConstructor(array &$data, $class, array &$context, \Reflec
      *
      * @throws RuntimeException
      */
-    protected function instantiateObject(array &$data, $class, array &$context, \ReflectionClass $reflectionClass, $allowedAttributes/*, $format = null*/)
+    protected function instantiateObject(array &$data, $class, array &$context, \ReflectionClass $reflectionClass, $allowedAttributes, ?string $format = null)
```

Comment:
That's a common misconception to me: there is absolutely *no* difference between the two.
Internally, the PHP engines forgets about the semantic you see as a human very very early in the PHP compilation process. We should not encourage this misunderstanding. There is *no* difference really, neither technically nor semantically That's just exactly the same.